cli/cli.go

```diff
@@ -50,9 +50,6 @@ Output build version information.
 
 var cockroachCmd = &cobra.Command{
 	Use: "cockroach",
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {
-		context.Addr = util.EnsureHostPort(context.Addr)
-	},
```

Comment:
Ah, this is where we ensure a port on `context.Addr`. I'm curious who you didn't just add a line here to ensure a port on `context.PGAddr`. You'd have to change `EnsureHostPort` to take a port to append if one isn't present. As it is, there is a lot of duplicated functionality between `EnsureHostPort` and your new `portPadding` function.